File: modules/Plugin/src/daybreak/abilitywar/utils/base/collect/PushingList.java
package daybreak.abilitywar.utils.base.collect;

import java.util.LinkedList;

public class PushingList<T> extends LinkedList<T> {

	private final int size;

	public PushingList(int size) {
		this.size = size;
	}

	@Override
	public boolean add(T e) {
		if (size() >= size) {
			poll();
		}
		return super.add(e);
	}

}

File: modules/Plugin/src/daybreak/abilitywar/utils/base/collect/PushingList.kt
package daybreak.abilitywar.utils.base.collect

import java.util.LinkedList

class PushingList<T>(private val maxSize: Int) : LinkedList<T>() {
	override fun add(e: T): Boolean {
		if (size >= maxSize) {
			poll()
		}
		return super.add(e)
	}

}